cognite/client/credentials.py

```diff
@@ -372,3 +372,68 @@ def _refresh_access_token(self) -> Tuple[str, float]:
             raise CogniteAuthError(
                 f"Error generating access token: {oauth_err.error}, {oauth_err.status_code}, {oauth_err.description}"
             ) from oauth_err
+
+
+class OAuthClientCertificate(_OAuthCredentialProviderWithTokenRefresh):
+    """OAuth credential provider for authenticating with a client certificate.
+
+    Args:
+        authority_url (str): OAuth authority url
+        client_id (str): Your application's client id.
+        cert_thumbprint (str): Your certificate's thumbprint. You get it when you upload your certificate to Azure AD.
+        certificate (str): Your private certificate, typically read from a .pem file
+        scopes (List[str]): A list of scopes.
+        **token_custom_args (Any): Optional additional arguments to pass as query parameters to the token fetch request.
+
+    Examples:
+
+            >>> from cognite.client.credentials import OAuthClientCertificate
+            >>> import os
+            >>> oauth_provider = OAuthClientCertificate(
+            ...     authority_url="https://login.microsoftonline.com/xyz",
+            ...     client_id="abcd",
+            ...     cert_thumbprint="XYZ123",
+            ...     certificate=open("certificate.pem").read(),
+            ...     scopes=["https://greenfield.cognitedata.com/.default"],
```

Comment:
Let's add a suggestion that explicitly closes the file (instead of relying on garbage collection). Also `os` seems to be unused.
```suggestion
            >>> from cognite.client.credentials import OAuthClientCertificate
            >>> from pathlib import Path
            >>> oauth_provider = OAuthClientCertificate(
            ...     authority_url="https://login.microsoftonline.com/xyz",
            ...     client_id="abcd",
            ...     cert_thumbprint="XYZ123",
            ...     certificate=Path("certificate.pem").read_text(),
            ...     scopes=["https://greenfield.cognitedata.com/.default"],
```

Reply:
Thanks for the blazing fast review and suggestions! Have fixed them now.